packages/timeline/src/ReactAxisGanttSeries.ts

```diff
@@ -0,0 +1,248 @@
+        .stroke("#000000")
+        .fitHeightToContent(true)
+        ;
+    protected _bottomAxis: Axis = new Axis("origin")
+        .orientation("bottom")
+        .type("linear")
+        .shrinkToFit("none")
+        .overlapMode("hide")
+        .extend(0)
+        .tickFormat("d")
+        ;
+    
+    protected _topAxisElement;
+    protected _contentElement;
+    protected _bottomAxisElement;
+    protected _topRect;
+    protected _contentRect;
+    protected _bottomRect;
+
+    protected rangeRenderer;
+
+    constructor(){
+        super();
+        this.centerOverflowX_default("hidden");
+        this.centerOverflowY_default("auto");
+
+        this.rangeRenderer = function () {
+            const ret = this._gantt.rangeRenderer.apply(this._gantt, arguments);
+            if(!arguments.length)return ret;
+            return this;
+        };
+    }
+
+    resizeWrappers() {
+
+        const w = this.width();
+        const h = this.height();
+
+        const axisHeight = this.axisHeight(); //TODO: Dynamic scaling to allow for small resolutions?
+        const contentHeight = (h - (axisHeight * 2)); 
+
+        this.bottomHeight(axisHeight);
+
+        this._topWA.resize({
+            width: w,
+            height: axisHeight
+        });
+        this._centerWA.resize({
+            width: w,
+            height: contentHeight
+        });
+        this._bottomWA.resize({
+            width: w,
+            height: axisHeight
+        });
+        this.top().render();
+        this.bottom().render();
+        this.center().render();
+    }
+
+    enter(domNode, element){
+        super.enter(domNode, element);
+
+        this._gantt.click = (row, col, sel) => {
+            this.click(row, col, sel);
+        };
+
+        this._gantt.dblclick = (row, col, sel) => {
+            this.dblclick(row, col, sel);
+        };
+
+        this.top(this._topAxis);
+        this.center(this._gantt);
+        this.bottom(this._bottomAxis);
+
+        this.resizeWrappers();
+
+        this._gantt.zoomedHook = (transform) => {
+            this.onzoom(transform);
+        };
+    }
+
+    onzoom(transform) {
+
+        const w = this.width();
+        const low = this._gantt._minStart;
+        const high = this._gantt._maxEnd;
+        const range = high - low;
+        const wpp = range / w;
+        const nextLow = Math.floor(low - (wpp * (transform.x / transform.k)));
+        const nextHigh = Math.ceil((range / transform.k) + nextLow);
+
+        this._topAxis
+            .fontFamily(this.axisFontFamily())
+            .fontSize(this.axisFontSize())
+            .tickLength(this.axisTickLength())
+            .low(nextLow)
+            .high(nextHigh)
+            .render()
+            ;
+        this._bottomAxis
+            .fontFamily(this.axisFontFamily())
+            .fontSize(this.axisFontSize())
+            .tickLength(this.axisTickLength())
+            .low(nextLow)
+            .high(nextHigh)
+            .render()
```

Comment:
Should this be a lazyRender?

Reply:
In this case lazyRender causes the axes to lag behind the rendering of the data points. Rapidly rendering the axes keeps them in sync with the data even with thousands of data rows.